modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Filters.tsx

```diff
@@ -73,33 +90,53 @@ interface IPropsAddFDSFilterModalContent {
 	closeModal: Function;
 	fdsView: FDSViewType;
 	fields: IField[];
-	filter?: IDateFilter | IDynamicFilter;
+	filter?: IDateFilter | ISelectionFilter;
+	fieldNames?: string[];
+	filterType?: filterTypes;
 	namespace: string;
 	onSave: (newFilter: IFilter) => void;
 }
 
+function getModalHeader(
+	filter: IFilter | undefined,
+	filterType: filterTypes | undefined
+): string {
+	if (filter) {
+		return sub(Liferay.Language.get('edit-x-filter'), [filter.name]);
+	}
+
+	if (filterType && filterType == filterTypes.SELECTION) {
+		return Liferay.Language.get('new-selection-filter');
+	} else {
+		return Liferay.Language.get('new-date-range-filter');
+	}
+}
```

Comment:
Do we really need to extract this to a helper function? We only call it once and could easily just inline these conditionals

Reply:
Done